### PR TITLE
Npm script exercise and date-fn tests

### PR DESCRIPTION
**Pull request by tong-lambda**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +25,10 @@
     "mocha": "8.2.1"
   },
   "dependencies": {
+    "date-fns": "2.17.0",
+    "dateformat": "4.5.1",
     "esm": "3.2.25",
-    "lodash": "4.17.20"
+    "express": "git+https://github.com/visionmedia/express.git",
```
Comment (tong-lambda) [inline package.json:31]: Not using it anywhere, just practice installing it.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,11 @@
   "main": "index.js",
   "scripts": {
     "test": "mocha --require esm",
-    "tw": "mocha --watch --watch-files 'test/**/*.js' --watch-files 'src/**/*.js' --reporter dot --require esm"
+    "tw": "mocha --watch --watch-files 'test/**/*.js' --watch-files 'src/**/*.js' --reporter dot --require esm",
```
Comment (josh-egan) [inline package.json:8]: Now that you know how to run one npm script from another, how could you modify the `tw` script to make use of this feature?

**Review by josh-egan — COMMENTED** [review]
Looking great! :clap: A few comments below.

File: README.md [hunk 3]
```diff
@@ -25,3 +25,31 @@
...
     https://regex101.com/
+
+## NPM
+
+- To install a package
+
+  - How to install as a dependency
+    `npm intall (-P or --save-prod is optional)`
+  - How to install as a dev dependency
+    `npm install - D or --save-dev`
```
[the remark this answers is not in the document]
Comment (tong-lambda) [reply README.md:36]: fixed

File: README.md [hunk 4]
```diff
@@ -25,3 +25,31 @@
...
+  - How to install from a GitHub repository
+    `npm install <github-tarball-url>`
+    example: `npm install git+https://github.com/visionmedia/express.git`
+
+- npm ci
+  `npm ci` will do the following things:
+
+  - It will delete your node_modules folder to ensure a clean state.
+  - It will look in your `package-lock.json` to install all the dependencies with the exact version.
+  - Unlike `npm install`, `npm ci` will never modify your `package-lock.json`. It does however expect a `package-lock.json` file in your project — if you do not have this file, `npm ci` will not work and you have to use `npm install`. instead.
```
[the remark this answers is not in the document]
Comment (tong-lambda) [reply README.md:48]: fixed

File: README.md [hunk 5]
```diff
@@ -25,3 +25,31 @@
...
+
+- Check outdated packages
+
+  - Run `npm outdated` or `npm outdated <package name>` for a specific pagage
```
[the remark this answers is not in the document]
Comment (tong-lambda) [reply README.md:52]: fixed

File: README.md [hunk 6]
```diff
@@ -25,3 +25,31 @@
...
+  - Run `npm update` to update the packages
```
[the remark this answers is not in the document]
Comment (tong-lambda) [reply README.md:53]: Thank you for the detiail, I'll put those in readme

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (josh-egan) [reply README.md:36]: FYI when a reviewer leaves a `Suggested change` comment, you can click the `Commit suggestion` button to make the change right from the GitHub UI.

**Review by josh-egan — APPROVED** [review]
Looks great!

**Pull request merged**